From 0d00c86f2926e51a1ff51d43f33b29ace25ffd5f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Sun, 10 Sep 2017 23:11:03 +0200
Subject: Code coverage is optional

Forwarded: no
---
 bin/run.js   | 22 +++++++++++++++++++---
 package.json |  2 +-
 2 files changed, 20 insertions(+), 4 deletions(-)

--- a/bin/run.js
+++ b/bin/run.js
@@ -45,13 +45,17 @@
 // Currently only Coveralls is supported, but the infrastructure
 // is left in place in case some noble soul fixes the codecov
 // module in the future.  See https://github.com/tapjs/node-tap/issues/270
-const coverageServices = [
+var coverageServices = []
+try {
+coverageServices.push(
   {
     env: 'COVERALLS_REPO_TOKEN',
     bin: require.resolve('coveralls/bin/coveralls.js'),
     name: 'Coveralls'
   }
-]
+);
+}
+catch(e) {}
 
 const main = _ => {
   const args = process.argv.slice(2)
@@ -94,7 +98,14 @@
   /* istanbul ignore next */
   if ((options.coverageReport || options.checkCoverage) &&
       options.files.length === 0)
+    {
+    try {
     return runCoverageReport(options)
+    }
+    catch(e) {
+        console.warn("Code coverage is disabled because some modules are missing")
+    }
+    }
 
   if (options.files.length === 0) {
     console.error('Reading TAP data from stdin (use "-" argument to suppress)')
@@ -112,8 +123,13 @@
   // By definition, the next block cannot be covered, because
   // they are only relevant when coverage is turned off.
   /* istanbul ignore next */
-  if (options.coverage && !global.__coverage__) {
+  if (options.coverage && !global.__coverage__)
+  {
+  try {
     return respawnWithCoverage(options)
+  } catch(e) {
+    console.warn("Code coverage is disabled because some modules are missing")
+  }
   }
 
   setupTapEnv(options)
--- a/package.json
+++ b/package.json
@@ -16,7 +16,6 @@
     "bluebird": "^3.5.1",
     "clean-yaml-object": "^0.1.0",
     "supports-color" : "^3",
-    "coveralls": "^3.0.1",
     "foreground-child": "^1.3.3",
     "fs-exists-cached": "^1.0.0",
     "function-loop": "^1.0.1",
@@ -40,6 +39,7 @@
   },
   "optionalDependencies" : {
     "nyc": "^11.8.0",
+    "coveralls": "^3.0.1",
     "opener": "^1.4.1"
   },
   "keywords": [
