Description: fix for foreground 3
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2023-08-17

--- a/bin/run.js
+++ b/bin/run.js
@@ -10,7 +10,7 @@
 const opener = require('opener')
 const node = process.execPath
 const fs = require('fs')
-const fg = require('foreground-child')
+const { foregroundChild } = require('foreground-child')
 const {spawn, spawnSync} = require('child_process')
 const nycBin = require.resolve(
   'nyc/' + require('nyc/package.json').bin.nyc
@@ -334,7 +334,7 @@
   args.push.apply(args, programArgs)
 
   if (spawnOpts)
-    return fg(node, args, spawnOpts)
+    return foregroundChild(node, args, spawnOpts)
 
   // fake it
   process.argv = [
@@ -400,7 +400,7 @@
   }
 }
 
-const nycHelp = () => fg(node, [nycBin, '--help'])
+const nycHelp = () => foregroundChild(node, [nycBin, '--help'])
 
 // export for easier testing
 const setupTapEnv = exports.setupTapEnv = options => {
--- a/module-libtap/npm-run-test.js
+++ b/module-libtap/npm-run-test.js
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 'use strict'
 
-const foregroundChild = require('foreground-child')
+const { foregroundChild } = require('foreground-child')
 
 async function runTests() {
   const semver = require('semver')
